arch/riscv/core/isr.S

```diff
@@ -356,7 +356,7 @@ no_fp:	/* increment _current->arch.exception_depth */
 	li t1, RISCV_EXC_ECALLU
 	beq t0, t1, is_user_syscall
 
-#ifdef CONFIG_PMP_STACK_GUARD
+#if defined(CONFIG_PMP_STACK_GUARD) && defined(MULTITHREADING)
```

Comment:
Shouldn't it be `CONFIG_MULTITHREADING` instead of `MULTITHREADING`?

Reply:
I took the liberty and fixed that